### PR TITLE
python3Packages.werkzeug,python3Packages.aiohttp,python3Packages.async-upnp-client: fix build; pythonPackages.urllib3: disable failing test 

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/115521

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -8522,7 +8526,13 @@ in {
 
   webthing = callPackage ../development/python-modules/webthing { };
 
-  werkzeug = callPackage ../development/python-modules/werkzeug { };
+  werkzeug = callPackage ../development/python-modules/werkzeug {
+    pytestCheckHook = self.pytestCheckHook.override { pytest = self.pytest_6_1; };
+    pytest-timeout = self.pytest-timeout.override {
+      pytest = self.pytest_6_1;
+      pytestCheckHook = self.pytestCheckHook.override { pytest = self.pytest_6_1; };
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix:8533]: How about creating `pytest6_1CheckHook`?

**Review by mweinelt — COMMENTED** [review]
Changes look reasonable.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -8522,7 +8526,13 @@ in {
 
   webthing = callPackage ../development/python-modules/webthing { };
 
-  werkzeug = callPackage ../development/python-modules/werkzeug { };
+  werkzeug = callPackage ../development/python-modules/werkzeug {
+    pytestCheckHook = self.pytestCheckHook.override { pytest = self.pytest_6_1; };
+    pytest-timeout = self.pytest-timeout.override {
+      pytest = self.pytest_6_1;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix:8532]: pytest plugins/extensions should not propagate pytest, see c44ae94eab1bcf4dbef5405800fc597d0b4f78dd

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (FRidh) [reply pkgs/top-level/python-packages.nix:8533]: Yes, or revert the pytest update. We should give packages time to update, although they've had half a year.

**Comment by mweinelt** [comment]
@SuperSandro2000 PTAL.

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/top-level/python-packages.nix:8533]: I don't like reverting stuff so the ``pytest_6_1CheckHook`` it is.

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -124,6 +124,8 @@ in {
     venvShellHook
     wheelUnpackHook;
 
+  pytest_6_1CheckHook = pytestCheckHook.override { pytest = pytest_6_1; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:127]: Not sure if I need a self here but ofborg will probably tell me.

**Comment by SuperSandro2000** [comment]
Splitted out the mass rebuild to https://github.com/NixOS/nixpkgs/pull/116115

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115564` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>4 packages failed to build and are new build failures:</summary>
  <ul>
<li>home-assistant: log was empty</li>
<li>python3Packages.async-upnp-client: log was empty</li>
<li>python3Packages.trio: log was empty</li>
<li>python3Packages.werkzeug: log was empty</li>
  </ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python3Packages.werkzeug:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/werkzeug/default.nix:24:3:

```
   |
24 |   meta = with lib; {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
warning: unused-argument
Unused argument: `isPy3k`.
Near pkgs/development/python-modules/werkzeug/default.nix:5:3:

```
  |
5 | , isPy3k
  |   ^
```
</details>
</details>

**Comment by SuperSandro2000** [comment]
I can't get urllib3 to pass the tests without disabling many tests.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115564` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python3Packages.async-upnp-client</li>
<li>python3Packages.trio</li>
<li>python3Packages.werkzeug</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115564` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>home-assistant</li>
<li>python3Packages.async-upnp-client</li>
<li>python3Packages.trio</li>
<li>python3Packages.werkzeug</li>
</ul>
</details>

File: pkgs/development/python-modules/urllib3/default.nix [hunk 4]
```diff
@@ -46,6 +46,10 @@ buildPythonPackage rec {
     trustme
   ];
 
+  disabledTests = [
+    "test_connection_closed_on_read_timeout_preload_false"
```
Comment (FRidh) [inline pkgs/development/python-modules/urllib3/default.nix:50]: fails only on aarch64. Disabling altogether will result in a mass rebuild.

Comment (FRidh) [reply pkgs/development/python-modules/urllib3/default.nix:50]:
```
disabledTests = if stdenv.hostPlatform.isAarch64 then [  "test_connection_closed_on_read_timeout_preload_false" ] else null;
```

Comment (FRidh) [reply pkgs/development/python-modules/urllib3/default.nix:50]: pushed 2e0b261da8c4a08d642bcbf115ffbca5d69ff481 to staging-next

**Comment by FRidh** [comment]
Rebased and pushed to staging-next, thank you.

File: pkgs/development/python-modules/urllib3/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/urllib3/default.nix:50]: I build urrlib3 on amd64 with --check and this was required to build the tests. I don't think we get around the mass rebuild.

**Pull request closed without merging**